cre.cpp

```diff
@@ -1311,6 +1353,107 @@ static int getWordBoxesFromPositions(lua_State *L) {
 	return 1;
 }
 
+static int getDocumentFileContent(lua_State *L) {
+    CreDocument *doc = (CreDocument*) luaL_checkudata(L, 1, "credocument");
+    const char* internalFilePath = luaL_checkstring(L, 2);
+
+    LVStreamRef stream = doc->text_view->getDocumentFileStream(lString16(internalFilePath));
+    if (!stream.isNull()) {
+        unsigned size = stream->GetSize();
+        lvsize_t read_size = 0;
+        void *buffer = (void *)malloc(size);
+        if (buffer != NULL) {
+            stream->Read(buffer, size, &read_size);
+            if (read_size == size) {
+                // (We can push a string containing NULL bytes with lua_pushlstring,
+                // so this function works with binary files like images.)
```

Comment:
Are you just saying that as a theoretical possibility?

Reply:
No, it's a practical reality :) I tested that with embedded png (it failed first, until I found the right way to do it).